pycheribuild/projects/disk_image.py

```diff
@@ -458,13 +476,33 @@ def add_all_files_in_dir(self, root_dir: Path):
 
     @property
     def is_x86(self):
-        return False
+        return self.crosscompile_target.is_any_x86()
 
     def run_mkimg(self, cmd: list, **kwargs):
         if not self.mkimg_cmd or not self.mkimg_cmd.exists():
             self.fatal("Missing mkimg command! Should be found in FreeBSD build dir (or set $MKIMG_CMD)")
         self.run_cmd([self.mkimg_cmd] + cmd, **kwargs)
 
+    @property
+    def include_efi_partition(self):
+        # TODO: Make this unconditional once all branches support EFI
+        if self.crosscompile_target.is_riscv(include_purecap=True):
+            return (self.rootfs_dir / "boot/loader.efi").exists()
+        return False
+
+    @property
+    def include_swap_partition(self):
+        return self.crosscompile_target.is_riscv(include_purecap=True)
+
+    @property
+    def rootfs_only(self):
```

Comment:
This function also seems to have a strangely-shaped predicate.  At least some docs would probably help, even if it has to stay strangely shaped and can't be pushed into child classes like `BuildMinimalCheriBSDDiskImage`?

Reply:
_sigh_ the RISC-V predicate is meant to be negated